### ISSUE TITLE
Equation variables in subcomponents do not resolve

### ISSUE THREAD

**Issue by kfhoech** [smaccm/smaccm#163]
Reported by @janetlj.  Thanks!

**Comment by kfhoech** [comment]
Duplicated by the following example:

~~~~~
package Issue_163_Pkg
public

	with Base_Types;

	system Inner
		features
			xx : in data port Base_Types::Integer;
			yy : out data port Base_Types::Integer;
		annex agree {**
			eq vv : int = xx + 1;
		**};
	end Inner;

	system implementation Inner.Impl
		annex agree {**
			eq ww : int = vv + 1;
			assign yy = ww + 1;
		**};
	end Inner.Impl;

	system Outer
		annex agree {**
			guarantee "Trivial" : true;
		**};
	end Outer;

	system implementation Outer.Impl
		subcomponents
			inner : system Inner;
			inner_impl : system Inner.Impl;
		annex agree {**
			eq ovv1 : int = inner.vv;
			eq oxx1 : int = inner.xx;
			eq oww1 : int = inner.ww; -- ww should not resolve in the type
			eq ovv2 : int = inner_impl.vv;
			eq oww2 : int = inner_impl.ww;
			eq oyy2 : int = inner_impl.yy;
		**};
	end Outer.Impl;

end Issue_163_Pkg;
~~~~~

After further investigation, looks like AGREE elements introduced by are missing.  Need to add call to collect these in scope provider.

**Comment by kfhoech** [comment]
Resolved by #166.
